plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -1123,9 +1123,11 @@ void chain_plugin::plugin_initialize(const variables_map& options) {
 
       my->accepted_block_connection = my->chain->accepted_block.connect( [this]( const block_state_ptr& blk ) {
          if (auto dm_logger = my->chain->get_deep_mind_logger()) {
+            auto packed_blk = fc::raw::pack(*blk);
```

Comment:
When a `shared_ptr` is used, it adds a first optionality byte to the binary data. So I had to use this here, but I'm wondering if this creates a copy of the structure prior packing it, I think it does.

Is there a way to avoid it if it's the case? Can I cast somehow to a `const block_state&` type?

Reply:
There is no copy of `*blk` before packing.  Also, `fc::raw::pack` takes its argument as `const T&`.

Reply:
So what's pointing to the `shared_ptr` gets cast as `const ref`, perfect thanks!